### DISCUSSION TITLE
how to get the output of the -compare command in batch?

### DISCUSSION THREAD

**Discussion by Sabeurre** [ImageMagick/ImageMagick#2734] [Help] [answered]
I would like to get the result of the compare command into a variable.
I tried:
for /f %%A in ('magick compare -metric PSNR 01.jpg 02.jpg compare.png') do set "myvar=%%A"
but it doesn't work
I also tried to output the result into a text file but it's empty
I'm stuck, any help would be very much appreciated.

**Comment by snibgo** [comment, marked as answer]
The number is written to stderr, not stdout, so we need to redirect stderr to stdout. In Windows BAT, we can retrieve it like this:
```
for /f %%A in ('magick compare -metric PSNR img1.png img2.png out.png 2^>^&1') do set myvar=%%A
```

**Reply by Sabeurre** [reply]
This always set myvar as "operable" what am I missing?

**Reply by t246246** [reply]
try to add -quiet.

**Reply by Sabeurre** [reply]
it works now, thank you

**Comment by snibgo** [comment]
What is the output of the simple command, ie:
```
magick compare -metric PSNR img1.png img2.png out.png
```
